pkg/providers/amifamily/ami.go

```diff
@@ -181,14 +203,11 @@ func (p *Provider) fetchAMIsFromSSM(ctx context.Context, ssmQuery string) (strin
 	return ami, nil
 }
 
-func (p *Provider) getAMIsFromSelector(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) ([]AMI, error) {
-	ec2AMIs, err := p.fetchAMIsFromEC2(ctx, nodeTemplate.Spec.AMISelector)
+func (p *Provider) getAMIsFromSelector(ctx context.Context, selector map[string]string) ([]AMI, error) {
+	ec2AMIs, err := p.fetchAMIsFromEC2(ctx, selector)
 	if err != nil {
 		return nil, err
 	}
-	if len(ec2AMIs) == 0 {
-		return nil, fmt.Errorf("no amis exist given constraints")
-	}
```

Comment:
It feels odd to remove this, just to add it back in on the other occurrences of amiProvider.Get() in other parts of the code. I think it makes more sense to keep this here.